server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -44,27 +56,82 @@ public void define(NewController context) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    JsonWriter json = response.newJsonWriter().beginObject();
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      Optional<UserDto> user = selectCurrentUser(dbSession);
+      Collection<String> groups = selectGroups(dbSession);
+      writeResponse(response, user, groups);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private Optional<UserDto> selectCurrentUser(DbSession dbSession) {
+    if (isNullOrEmpty(userSession.getLogin())) {
+      return Optional.absent();
+    }
 
-    writeUserDetails(json, userSession);
+    return Optional.fromNullable(dbClient.userDao().selectActiveUserByLogin(dbSession, userSession.getLogin()));
+  }
+
+  private Collection<String> selectGroups(DbSession dbSession) {
+    String login = userSession.getLogin();
+    if (isNullOrEmpty(login)) {
+      return emptyList();
+    }
+
+    Multimap<String, String> groupsByLogins = dbClient.groupMembershipDao().selectGroupsByLogins(dbSession, singletonList(login));
+    return groupsByLogins.get(login);
+  }
 
+  private void writeResponse(Response response, Optional<UserDto> user, Collection<String> groups) {
+    JsonWriter json = response.newJsonWriter().beginObject();
+    writeUserDetails(json, user, groups);
     json.endObject().close();
   }
 
-  private void writeUserDetails(JsonWriter json, UserSession session) {
-    json.prop("isLoggedIn", session.isLoggedIn())
-      .prop("login", session.getLogin())
-      .prop("name", session.getName());
-    writePermissions(json, session);
+  private void writeUserDetails(JsonWriter json, Optional<UserDto> optionalUser, Collection<String> groups) {
+    json
+      .prop("isLoggedIn", userSession.isLoggedIn())
+      .prop("login", userSession.getLogin())
+      .prop("name", userSession.getName());
+    if (optionalUser.isPresent()) {
+      UserDto user = optionalUser.get();
+      json.prop("email", user.getEmail());
```

Comment:
with default user "admin", I get JSON response `"email": "",`. Our standard is to have absent field instead of blank isn't it ?